R/batchmark.R

```diff
@@ -73,14 +73,14 @@ getAlgoFun = function(lrn, measures, models, keep.extract) {
   force(lrn)
   force(measures)
   force(models)
-  function(job, data, instance, keep.extract = keep.extract) {
-    if (isTRUE(keep.extract)) {
-      extract.this = getExtractor(lrn)
-    } else {
-      extract.this = function(model) {
-        NULL
-      }
+  if (isTRUE(keep.extract)) {
+    extract.this = getExtractor(lrn)
+  } else {
+    extract.this = function(model) {
+      NULL
     }
+  }
+  function(job, data, instance) {
```

Comment:
Shouldn't we pass `extract.this` along here?

Reply:
No, it lives now in the Environment of the function we are returning. Together with `lrn`, `measures` and `models`.